nkms/client.py

```diff
@@ -55,13 +78,35 @@ def encrypt_key(self, key, pubkey=None, path=None, algorithm=None):
         :return: Encrypted key(s)
         :rtype: bytes
         """
-        pass
+        if not pubkey:
+            pubkey = self._pub_key
 
-    def decrypt_key(self, key, pubkey=None, path=None, owner=None):
+        if type(path) is tuple and len(path) > 1:
+            enc_keys = []
```

Comment:
Does this make more sense as a `set` to enforce uniqueness?

Reply:
Probably not - the order is important

Reply:
And uniqueness here is more like "if I used the same path, it means it's the same file"

Reply:
Right - of course.  

Another interesting option - does it make sense for it to be a dict, where the keys are paths?

Reply:
Well, that is a very good point. Not necessarily we want to show what the path is to everyone, but maybe if it's an (ordered) dict {hash(subpath) -> key}, that could be valuable!

Reply:
Is there a security implication to have the path in the clear as a key?  They'll have access to the path anyway at run time by dint of this function.

Reply:
...and it might be useful to be able to easily get a iterator of paths with enc_keys.keys().

Reply:
The security professional in me wants to think that this isn't a very good idea. I'll look into better structuring this, but in a secure manner.

Reply:
Can you help me understand the security implications here?  If it's a problem to use the path as a dict key, then it's a problem to use the path at all, isn't it?  IE, why do we even have the path crossing the wire in the clear in the first place?